packages/pipeline/src/scripts/pull_paradex_orderbook_snapshots.ts

```diff
@@ -0,0 +1,93 @@
+// tslint:disable:no-console
+import { BigNumber } from '@0x/utils';
+import * as R from 'ramda';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+
+import {
+    PARADEX_SOURCE,
+    ParadexActiveMarketsResponse,
+    ParadexMarket,
+    ParadexSource,
+    ParadexTokenInfo,
+    ParadexTokenInfoResponse,
+} from '../data_sources/paradex';
+import { TokenOrderbookSnapshot as TokenOrder } from '../entities';
+import * as ormConfig from '../ormconfig';
+import { parseParadexOrders } from '../parsers/paradex_orders';
+import { handleError } from '../utils';
+
+// Number of orders to save at once.
+const BATCH_SAVE_SIZE = 1000;
+
+let connection: Connection;
+
+(async () => {
+    connection = await createConnection(ormConfig as ConnectionOptions);
+    const apiKey = process.env.PARADEX_API_KEY;
+    if (apiKey === undefined) {
+        throw new Error('Missing required env var: PARADEX_API_KEY');
+    }
+    const paradexSource = new ParadexSource(apiKey);
+    const tokenOrderRepository = connection.getRepository(TokenOrder);
+    const markets = await paradexSource.getActiveMarketsAsync();
+    const tokenInfoResponse = await paradexSource.getTokenInfoAsync();
+    const extendedMarkets = addTokenAddresses(markets, tokenInfoResponse);
+    await Promise.all(
+        extendedMarkets.map(async (market: ParadexMarket) => getAndSaveMarketOrderbook(paradexSource, market)),
+    );
+    process.exit(0);
+})().catch(handleError);
+
+/**
+ * Extend the default ParadexMarket objects with token addresses.
+ * @param markets An array of ParadexMarket objects.
+ * @param tokenInfoResponse An array of ParadexTokenInfo containing the addresses.
+ */
+function addTokenAddresses(
+    markets: ParadexActiveMarketsResponse,
+    tokenInfoResponse: ParadexTokenInfoResponse,
+): ParadexMarket[] {
+    const extractSymbolAddress = (acc: Map<string, string>, tokenInfo: ParadexTokenInfo): Map<string, string> => {
+        acc.set(tokenInfo.symbol, tokenInfo.address);
+        return acc;
+    };
+
+    const symbolAddressMapping = R.reduce(extractSymbolAddress, new Map(), tokenInfoResponse);
+
+    markets.forEach((market: ParadexMarket) => {
+        if (symbolAddressMapping.has(market.baseToken)) {
+            market.baseTokenAddress = symbolAddressMapping.get(market.baseToken);
+        } else {
+            console.log(`${market.baseToken}: No address found.`);
```

Comment:
How serious/unexpected is this error condition? Should we replace `console.log` with either `console.error` or `throw new Error('your message here')`?

Reply:
Hm. I feel like in this case it's an inconsistency of the external datasource, not an unexpected result of our system. But I don't feel strongly. I'd be happy to change it if you prefer.

Reply:
In that case it should either be console.error or nothing. If it happens frequently and we know how to handle it, just don't log anything. If we don't have a good way of handling it, console.error is more appropriate.